File: src/main/java/com/temportalist/scanner/client/ProxyClient.java
package com.temportalist.scanner.client;

import com.temportalist.scanner.client.gui.GuiDecomposer;
import com.temportalist.scanner.common.ProxyCommon;
import com.temportalist.scanner.common.TEDecomposer;
import com.temportalist.scanner.common.inventory.ContainerDecomposer;
import cpw.mods.fml.client.registry.RenderingRegistry;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;

/**
 * @author TheTemportalist
 */
public class ProxyClient extends ProxyCommon {

	@Override
	public void register() {
		RenderingRegistry.registerBlockHandler(new ThermalHandler());
	}

	@Override
	public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x,
			int y, int z) {
		TileEntity tile = world.getTileEntity(x, y, z);
		if (tile != null) {
			if (ID == 0 && tile instanceof TEDecomposer) {
				return new GuiDecomposer(
						(ContainerDecomposer) this.getServerGuiElement(ID, player, world, x, y, z)
				);
			}
		}
		return super.getClientGuiElement(ID, player, world, x, y, z);
	}

}

File: src/main/scala/com/temportalist/thaumicexpansion/client/ProxyClient.scala
package com.temportalist.thaumicexpansion.client

import com.temportalist.thaumicexpansion.common.ProxyCommon
import com.temportalist.thaumicexpansion.common.container.ContainerAnalyzer
import com.temportalist.thaumicexpansion.common.init.TECBlocks
import com.temportalist.thaumicexpansion.common.tile.{TEAnalyzer, TEApparatus}
import cpw.mods.fml.client.registry.ClientRegistry
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.Item
import net.minecraft.tileentity.TileEntity
import net.minecraft.world.World
import net.minecraftforge.client.MinecraftForgeClient

/**
 *
 *
 * @author TheTemportalist
 */
class ProxyClient extends ProxyCommon {

	override def register(): Unit = {
		val ran: RenderAnalyzer = new RenderAnalyzer()
		ClientRegistry.bindTileEntitySpecialRenderer(classOf[TEAnalyzer], ran)
		MinecraftForgeClient.registerItemRenderer(
			Item.getItemFromBlock(TECBlocks.analyzer), ran)
		ClientRegistry.bindTileEntitySpecialRenderer(classOf[TEApparatus], RenderApparatus)
		MinecraftForgeClient.registerItemRenderer(
			Item.getItemFromBlock(TECBlocks.apparatus), RenderApparatus)
	}

	override def getClientElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int,
			z: Int, tileEntity: TileEntity): AnyRef = {
		if (tileEntity != null) tileEntity match {
			case te: TEAnalyzer =>
				return new GuiAnalyzer(new ContainerAnalyzer(te, player))
			case _ =>
		}
		null
	}

}
